core/src/main/java/io/substrait/expression/proto/ProtoExpressionConverter.java

```diff
@@ -250,6 +219,80 @@ public Expression from(io.substrait.proto.Expression expr) {
     };
   }
 
+  public ImmutableExpression.WindowFunctionInvocation fromWindowFunction(
+      io.substrait.proto.Expression.WindowFunction windowFunction) {
+    var functionReference = windowFunction.getFunctionReference();
+    var declaration = lookup.getWindowFunction(functionReference, extensions);
+    var argVisitor = new FunctionArg.ProtoFrom(this, protoTypeConverter);
+
+    var args =
+        fromFunctionArgumentList(
+            windowFunction.getArgumentsCount(),
+            argVisitor,
+            declaration,
+            windowFunction::getArguments);
+    var partitionExprs =
+        windowFunction.getPartitionsList().stream().map(this::from).collect(Collectors.toList());
+    var sortFields =
+        windowFunction.getSortsList().stream()
+            .map(this::fromSortField)
+            .collect(Collectors.toList());
+    var options =
+        windowFunction.getOptionsList().stream()
+            .map(this::fromFunctionOption)
+            .collect(Collectors.toMap(FunctionOption::getName, Function.identity()));
```

Comment:
Oooh, nice addition. I hadn't noticed we weren't extraction and converting the functions options ✨